File: app/src/main/java/localhost/toolkit/text/ErrorListenerList.java
package localhost.toolkit.text;

import java.util.ArrayList;

public class ErrorListenerList extends ArrayList<AbstractErrorListener> {
	public boolean matches(boolean hidden) {
		boolean matches = true;
		for (AbstractErrorListener editTextErrorListener : this)
			matches = editTextErrorListener.matches(hidden) && matches;
		return matches;
	}
}

File: app/src/main/java/localhost/toolkit/text/ErrorListenerList.kt
package localhost.toolkit.text

class ErrorListenerList : ArrayList<AbstractErrorListener>() {
    fun matches(showError: Boolean): Boolean {
        var matches = true
        forEach { matches = it.matches(showError) && matches }
        return matches
    }

    fun matches(): Boolean {
        var matches = true
        forEach { matches = it.matches() && matches }
        return matches
    }
}